### ISSUE TITLE
[$1000]  Emoji size from email reply is smaller then emoji size sent via normal chat and emoji does not use user selected skin color

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14753]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app and login with user A
2. Open user B chat and send a message
3. Open user B email
4. Reply to user A message by email and send only emojis

## Expected Result:
Size of emojis sent via email reply should be same as size of emoji sent via normal chat, emoji skin color should also match to the one used by the user

## Actual Result:
Size of emojis is smaller then size of emojis sent via normal chat and skin color too does not match

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.2.63-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/216351996-a259a36c-cd7d-4260-9e78-79d3714f275e.mp4

https://user-images.githubusercontent.com/43996225/216352049-c590c637-ddfb-46e4-9696-e002ff6bb357.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675272531555829

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0174e50816828c7960</li>
        <li>Upwork Job ID: 1622671435996983296</li>
        <li>Last Price Increase: 2024-12-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @pecanoro</details>

**Comment by mallenexpensify** [comment]
Testing, not getting emails for all chats sent from my test account.

**Comment by MelvinBot** [comment]
@mallenexpensify Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mallenexpensify** [comment]
ah ha! Was able to reproduce, nice find @dhanashree-sawant 
Pretty sure this can be external.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0174e50816828c7960

**Comment by MelvinBot** [comment]
Current assignee @mallenexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @pecanoro (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/pages/home/report/ReportActionItemFragment.js b/src/pages/home/report/ReportActionItemFragment.js
index c4f593517d..b4ced62e10 100644
--- a/src/pages/home/report/ReportActionItemFragment.js
+++ b/src/pages/home/report/ReportActionItemFragment.js
@@ -104,9 +104,10 @@ const ReportActionItemFragment = (props) => {
             // we render it as text, not as html.
             // This is done to render emojis with line breaks between them as text.
             const differByLineBreaksOnly = Str.replaceAll(html, '<br />', '\n') === text;
+            const differByDivTagsOnly = Str.replaceAll(html, /<div[^>]*>|<\/div>/g, '') === text;
 
             // Only render HTML if we have html in the fragment
-            if (!differByLineBreaksOnly) {
+            if (!differByLineBreaksOnly && !differByDivTagsOnly) {
                 const editedTag = props.fragment.isEdited ? '<edited></edited>' : '';
                 const htmlContent = html + editedTag;
                 return (
```

### RCA
Sending messages from email will most likely be sent as html and that's the case here where emojis are sent as `<div dir="ltr">😁</div>`. Since the message is technically html it will be rendered using our HTML render engine. However as you can see there is nothing special about that message to treat it as html and we can instead just treat as a text.

We follow this approach in another similar case where the difference between the html message and the plaintext message is just line breaks (`differByLineBreaksOnly`) and I propose is to do the same here by checking if the difference between the two messages is just those "useless" div tags (`differByDivTagsOnly`).

So, we strip the html message from those useless div tags and if it's the same as the text then just treat it as text.

___

- We can extend the regex to cover other "useless" tags such as `span`, `p`, etc.
- Why those tags are "useless"?  because they produce no difference in the final markdown format.

**Comment by bernhardoj** [comment]
I think this issue will eventually be fixed by [this](https://github.com/Expensify/App/issues/4114) improvement.

**Comment by pecanoro** [comment]
@bernhardoj I think you are right. That issue encompasses a lot of fixes for emojis by making them consistent so I am going to add a HOLD on this one in benefit of that one. @mananjadhav What do you think? Or do you think we should fix this one separately?

**Comment by mananjadhav** [comment]
I can't make out if the email issue will be resolved, but we should hold this one anyway. The linked issue doesn't have any proposals so far, and I am fairly confident proposals will include some logic to match emoji, etc. 

So yeah let's hold this one @pecanoro

**Comment by pecanoro** [comment]
Still on HOLD

**Comment by mallenexpensify** [comment]
The updated PR link we're waiting on is https://github.com/Expensify/App/pull/15611
(long story)

**Comment by pecanoro** [comment]
Still waiting on https://github.com/Expensify/App/pull/15611

**Comment by pecanoro** [comment]
Still waiting on https://github.com/Expensify/App/pull/15611

**Comment by pecanoro** [comment]
Still waiting on https://github.com/Expensify/App/pull/15611

**Comment by pecanoro** [comment]
Still waiting on https://github.com/Expensify/App/pull/15611

**Comment by pecanoro** [comment]
Posted [here](https://github.com/Expensify/App/pull/15611#issuecomment-1587728224) asking why the PR is on HOLD to see what we should do.

**Comment by pecanoro** [comment]
Still waiting on https://github.com/Expensify/App/pull/15611

**Comment by mallenexpensify** [comment]
Till holding on https://github.com/Expensify/App/pull/15611

**Comment by mallenexpensify** [comment]
We're waiting on
https://github.com/Expensify/App/pull/15611
which is waiting on 
https://github.com/facebook/hermes/issues/932#issuecomment-1714399174
which will be live in v.73
https://reactnative.dev/versions

**Comment by mallenexpensify** [comment]
^ Same

**Comment by pecanoro** [comment]
Ohh it seems we finally have a new version! I am going to post in the PR!

**Comment by pecanoro** [comment]
Posted again [here](https://github.com/Expensify/App/pull/15611#issuecomment-1903879957) to figure out the next steps.

**Comment by mananjadhav** [comment]
The held issue was now assigned and I've posted a comment [here](https://github.com/Expensify/App/issues/4114#issuecomment-1930672554).

**Comment by mallenexpensify** [comment]
Issue we're held on is actively being worked on

**Comment by mallenexpensify** [comment]
@mananjadhav can you take a look here?
- https://github.com/Expensify/App/issues/4114#issuecomment-2016603948

I'm wondering if we can take this off hold, it looks like the associated PR was merged

**Comment by mananjadhav** [comment]
@mallenexpensify The associated PR is not merged. https://github.com/Expensify/App/pull/37980 I think the linked issue's PR was dependent on the one you linked in the previous comment. This should still be on hold.

**Comment by mallenexpensify** [comment]
Thanks @mananjadhav , updated the title to correctly reflect the PR we're held on.

**Comment by pecanoro** [comment]
Same update as last time, the PR is still in progress

**Comment by pecanoro** [comment]
Ohh https://github.com/Expensify/App/pull/40692 is close to getting merged!

**Comment by pecanoro** [comment]
https://github.com/Expensify/App/pull/40692 was finally merged! @mallenexpensify Could you check if this is still an issue or we can finally close it?

**Comment by pecanoro** [comment]
Oh nvm, it was reverted and now we are waiting on this: https://github.com/Expensify/App/pull/47547

**Comment by mallenexpensify** [comment]
Hold still

**Comment by pecanoro** [comment]
Hold still

**Comment by mallenexpensify** [comment]
Hit staging two hours ago
https://github.com/Expensify/App/pull/47547#issuecomment-2494746041

**Comment by mananjadhav** [comment]
The held PR is now deployed to production 2 days ago, can we unblock this now?

**Comment by mallenexpensify** [comment]
Attempting to retest, added `retest-weekly`, made me the owner and updated to `Weekly`.  
@mananjadhav are you able to reproduce?

**Comment by mananjadhav** [comment]
I retested this and it's still reproducible. 

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/c9b1d63d-1cb6-4a32-ad3d-bad03341c372">

**Comment by mallenexpensify** [comment]
@mananjadhav the two emoji appear to be the same size and color, am I missing something?

**Comment by mananjadhav** [comment]
@mallenexpensify  The skintone is fine, the size of the emoji should've been bigger. Checkout this video.

https://github.com/user-attachments/assets/beef310f-559c-4208-bc42-ba99c68561d5

**Comment by mananjadhav** [comment]
@mallenexpensify I think this would be internal. Can we get someone from the engineering team to confirm?

1. When we send the messages from our app, we render the message HTML as: `<emoji>\ud83d\udc4d\ud83c\udffe<\/emoji>`.

2. When the emoji is sent from the email the HTML is `<div dir=\"ltr\">\ud83d\udc4d\ud83c\udffe<\/div>`.

This message comes from the backend, so ideally we should be wrapping them with `emoji` tag. If you notice, the emoji unicode is same.

**Comment by mallenexpensify** [comment]
ah... got it! Thx
<img width="351" alt="image" src="https://github.com/user-attachments/assets/874382b2-f15a-4665-9ad2-8b7b02be2f00" />

**Comment by mananjadhav** [comment]
@pecanoro Can you take a look at this issue?

**Comment by mallenexpensify** [comment]
@pecanoro , can you confirm this is a backend bug? Thx

**Comment by pecanoro** [comment]
I am going to move to monthly because it's an edge case and I have a few more important internal ones to handle first

**Comment by pecanoro** [comment]
I will test the skin color tomorrow to see what's happening there

**Comment by mananjadhav** [comment]
@pecanoro Any updates on this?

**Comment by pecanoro** [comment]
The tone seems to work well, it's only the size:

![Image](https://github.com/user-attachments/assets/1b4d68b0-3a9c-48d4-9b9d-85d89405f031)

I could ask internally to see if anyone has a better idea on how to fix this, I couldn't figure it out after a decent a mount of time and since it's really low priority, I feel I should be focusing on other stuff

**Comment by mananjadhav** [comment]
There were two promising proposals earlier. Should we try to fix it on the frontend to render the emojis without the custom `<emoji>` tags?

**Comment by pecanoro** [comment]
We could also try that or do nothing, @mallenexpensify what do you think? Both solutions are a bit weird since it's gmail the one wrapping the emojis in divs.

**Comment by mallenexpensify** [comment]
@mananjadhav how about posting in #expensify-open-source to get more 👀 and feedback on our options?  I don't love fixing it only for gmail but...  so many people use gmail.

**Comment by mallenexpensify** [comment]
@mananjadhav , @s77rt [replied in thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1736962742407189?thread_ts=1736958989.235039&cid=C01GTK53T8Q)
> I'm still inclined for the frontend approach and use [shouldRenderAsText](https://github.com/Expensify/App/blob/28deaadd5ebb1bcafa34111e61366a3816b2d79f/src/pages/home/report/comment/shouldRenderAsText/index.ts#L5) as we'd be using an already existing solution on how we handle such cases. Otherwise :donothing:.

> FWIW, I don't think this should be fixed in the BE because I suspect that the BE does not do any text processing on the email content and add that processing just for this case is not worth it. However IF the BE does process email text then fixing this in BE is probably best.

Do you agree?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by s77rt** [comment]
This is not reproducible from my end too

**Comment by pecanoro** [comment]
Oh wow yes, it has been magically fixed lol

**Comment by pecanoro** [comment]
@mallenexpensify If you can't reproduce it either, let's close it!

**Comment by mallenexpensify** [comment]
Wow, I'll take magic fixes anytime!  Thx y'all

**Comment by mallenexpensify** [comment]
@mananjadhav due $500 via NewDot for C+ work on this issue, ie. 50% of job price. As discussed [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1738089140171669)

**Comment by mananjadhav** [comment]
Requested on NewDot.

**Comment by garrettmknight** [comment]
$500 approved for @mananjadhav
